tools/c7n_azure/c7n_azure/storage_utils.py

```diff
@@ -74,8 +74,8 @@ def get_storage_account_by_name(storage_account_name):
         return matching_account[0]
 
     @staticmethod
-    def get_storage_keys(storage_account_id):
-        s = local_session(Session)
+    def get_storage_keys(storage_account_id, session=None):
+        s = session or local_session(Session)
```

Comment:
You could do the following: `session=local_session(Session)` rather than `s = session or local_session(Session)
 `

That being said... I find it a little confusing when a helper function actually fetches the session via
`s = session or local_session(Session)` versus just used the passed parameter. 

I may have missed something, but could we make it all consistent?

Reply:
So it needs to work with and without session coming in - so I'm just doing a null coalesce to use the provided value if it isn't empty, otherwise make on.

Storage utils get used a bunch of places outside of the context of a policy and in other tools (mailer, etc) so I wanted to leave it backward compatible where it would make a naive session up if one wasn't passed in - but if someone who has `ExecutionContext` is using it they need to be able to pass one in.